cmd/brew-test-bot.rb

```diff
@@ -977,7 +980,10 @@ def test_ci_upload(tap)
 
     if (pr = ENV["UPSTREAM_PULL_REQUEST"])
       pull_pr = "https://github.com/#{tap.user}/homebrew-#{tap.repo}/pull/#{pr}"
-      safe_system "brew", "pull", "--clean", pull_pr
+      args = []
+      args << "--tap=#{@tap}" unless @tap.nil?
+      args << pull_pr
+      test "brew", "pull", "--clean", *args
```

Comment:
A supported feature of `brew tap a/b http://github.com/c/homebrew-d` is that the git remote may be custom. When that's the case, to use the `brew pull` command, currently the tap needs to be specified, because it cannot be inferred from the pull request URL.

For example
```sh
brew tap a/b http://github.com/c/homebrew-d
brew pull --tap=a/b http://github.com/c/homebrew-d/pull/123
```

When the git remote of a tap is custom, how does one use the `brew pull` command with that tap?